apps/src/studio/Item.js

```diff
@@ -358,3 +444,37 @@ Item.prototype.moveToNextPosition = function () {
   this.x = next.x;
   this.y = next.y;
 };
+
+/**
+ * Mark that we're colliding with object represented by key.
+ * Here, override base implemention to special on-collision logic for certain
+ * item classes.
+ * @param key A unique key representing the object we're colliding with
+ * @returns {boolean} True if collision is started, false if we're already colliding
+ * @override
+ */
+Item.prototype.startCollision = function (key) {
+  var newCollisionStarted = Item.superPrototype.startCollision.call(this, key);
+  if (newCollisionStarted) {
+    if (this.className === 'hazard') {
```

Comment:
Would like to see this be more configurable (vs. hardcoding a specific className).
Ideas include:
* Introduce a level.failWhenActorCollidesWithItem boolean (would apply to all classes)
* Create a list of classNames on the skin or level that will trigger this behavior
* Extend `skin.specialItemProperties` to flag this behavior with a new property
* Leverage the existing `activity` field on `Item` to imply this behavior (I don't love this idea, but for now, it could be tied to the `watchActor` activity)